Prog/2022/Sales_affordability_2016_20.sas

```diff
@@ -0,0 +1,383 @@
+	if sale_yr=2020 then PI_First2020=saleprice*.9*loan_multiplier_2020;
+
+	%dollar_convert(PI_first2016,PI_first2016r,2016,2020, series=CUUR0000SA0L2);
+	%dollar_convert(PI_first2017,PI_first2017r,2017,2020, series=CUUR0000SA0L2);
+ 	%dollar_convert(PI_first2018,PI_first2018r,2018,2020, series=CUUR0000SA0L2);
+	%dollar_convert(PI_first2019,PI_first2019r,2019,2020, series=CUUR0000SA0L2);
+	%dollar_convert(PI_first2020,PI_first2020r,2020,2020, series=CUUR0000SA0L2);
+
+  *calculate monthly PITI (Principal, Interest, Taxes and Insurance) for First Time Homebuyer (34% of PI = TI);
+	if sale_yr=2016 then PITI_First=PI_First2016r*1.34;
+	if sale_yr=2017 then PITI_First=PI_First2017r*1.34;
+	if sale_yr=2018 then PITI_First=PI_First2018r*1.34;
+	if sale_yr=2019 then PITI_First=PI_First2019r*1.34;
+	if sale_yr=2020 then PITI_First=PI_First2020r*1.34;
+
+  *calculate monthly Principal and Interest for Repeat Homebuyer (20% down);
+    if sale_yr=2016 then PI_Repeat2016=saleprice*.8*loan_multiplier_2016;
+	if sale_yr=2017 then PI_Repeat2017=saleprice*.8*loan_multiplier_2017;
+	if sale_yr=2018 then PI_Repeat2018=saleprice*.8*loan_multiplier_2018;
+	if sale_yr=2019 then PI_Repeat2019=saleprice*.8*loan_multiplier_2019;
+	if sale_yr=2020 then PI_Repeat2020=saleprice*.8*loan_multiplier_2020;
+
+	%dollar_convert(PI_Repeat2016,PI_Repeat2016r,2016,2020,series=CUUR0000SA0L2);
+	%dollar_convert(PI_Repeat2017,PI_Repeat2017r,2017,2020,series=CUUR0000SA0L2);
+ 	%dollar_convert(PI_Repeat2018,PI_Repeat2018r,2018,2020,series=CUUR0000SA0L2);
+	%dollar_convert(PI_Repeat2019,PI_Repeat2019r,2019,2020,series=CUUR0000SA0L2);
+	%dollar_convert(PI_Repeat2020,PI_Repeat2020r,2020,2020,series=CUUR0000SA0L2);
+
+	*calculate monthly PITI (Principal, Interest, Taxes and Insurance) for Repeat Homebuyer (25% of PI = TI);
+	if sale_yr=2016 then PITI_Repeat=PI_Repeat2016r*1.25;
+	if sale_yr=2017 then PITI_Repeat=PI_Repeat2017r*1.25;
+	if sale_yr=2018 then PITI_Repeat=PI_Repeat2018r*1.25;
+	if sale_yr=2019 then PITI_Repeat=PI_Repeat2019r*1.25;
+	if sale_yr=2020 then PITI_Repeat=PI_Repeat2020r*1.25;
+
+
+	/*Here are numbers for Average Household Income at the city level. 2016-20 ACS 
+	Using tables B19025(B,H, I) and B11001X (B,H,I)
+	Black	NH-White	Hispanic	AIOM	 
+	72915	 194743		120441 	 	 		*/
+
+
+	if PITI_First <= (194743 / 12*.28) then white_first_afford=1; else white_first_afford=0; 
+		if PITI_Repeat <= (194743/ 12 *.28) then white_repeat_afford=1; else white_repeat_afford=0; 
+	if PITI_First <= (72915 / 12 *.28) then black_first_afford=1; else black_first_afford=0; 
+		if PITI_Repeat <= (72915 / 12 *.28) then black_repeat_afford=1; else black_repeat_afford=0; 
+	if PITI_First <= (120441 / 12*.28) then hispanic_first_afford=1; else hispanic_first_afford=0; 
+		if PITI_Repeat <= (120441 / 12*.28 ) then hispanic_repeat_afford=1; else hispanic_repeat_afford=0; 
+	/*if PITI_First <= (76271 / 12*.28 ) then aiom_first_afford=1; else aiom_first_afford=0; 
```

Comment:
@rpitingolo AIOM is commented out because i didn't have time to look them all up and add together. given the need for the Black homeownership strikeforce I'm leaving this alone for now.